pkg/clusteragent/admission/mutate/autoinstrumentation/configuration_cache.go

```diff
@@ -0,0 +1,289 @@
+		clusterName:                  clusterName,
+		namespaceToConfigIDMap:       nsToRules,
+
+		orderedRevisions: make([]int64, 0),
+		enabledRevisions: map[int64]enablementConfig{},
+	}
+}
+
+func (c *instrumentationConfigurationCache) start(stopCh <-chan struct{}) {
+	for {
+		select {
+		case req := <-c.configurationUpdatesQueue:
+			// if err := c.updateConfiguration(nil, nil); err != nil {
+			// 	log.Error(err.Error())
+			// }
+			resp := c.update(req)
+			c.configurationUpdatesResponse <- resp
+		case <-stopCh:
+			log.Info("Shutting down patcher")
+			return
+		}
+	}
+}
+
+func (c *instrumentationConfigurationCache) update(req Request) Response {
+
+	k8sClusterTargets := req.K8sTargetV2.ClusterTargets
+	var resp Response
+
+	switch req.Action {
+	case EnableConfig:
+		for _, target := range k8sClusterTargets {
+			clusterName := target.ClusterName
+
+			if c.clusterName == clusterName {
+				log.Infof("Current configuration: %v, %v, %v",
+					c.currentConfiguration.enabled, c.currentConfiguration.enabledNamespaces, c.currentConfiguration.disabledNamespaces)
+				newEnabled := target.Enabled
+				newEnabledNamespaces := target.EnabledNamespaces
+
+				c.mu.Lock()
+				resp = c.updateConfiguration(*newEnabled, newEnabledNamespaces, req.ID, int(req.RcVersion))
+				log.Infof("Updated configuration: %v, %v, %v",
+					c.currentConfiguration.enabled, c.currentConfiguration.enabledNamespaces, c.currentConfiguration.disabledNamespaces)
+
+				c.orderedRevisions = append(c.orderedRevisions, req.Revision)
+				c.enabledRevisions[req.Revision] = enablementConfig{
+					configID:          req.ID,
+					rcVersion:         int(req.RcVersion),
+					rcAction:          string(req.Action),
+					env:               req.LibConfig.Env,
+					enabled:           target.Enabled,
+					enabledNamespaces: target.EnabledNamespaces,
+				}
+				c.mu.Unlock()
+			}
+		}
+	default:
+		log.Errorf("unknown action %q", req.Action)
+	}
+
+	return resp
+}
+
+func (c *instrumentationConfigurationCache) delete(rcConfigID string) error {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+
+	for i, rev := range c.orderedRevisions {
+		confID, ok := c.enabledRevisions[rev]
+		if !ok {
+			log.Error("Revision was not found")
+		}
+		if confID.configID == rcConfigID {
+			delete(c.enabledRevisions, rev)
+			c.orderedRevisions = append(c.orderedRevisions[:i], c.orderedRevisions[i+1:]...)
+			break
+		}
+	}
+	c.resetConfiguration()
+	return nil
+}
+
+func (c *instrumentationConfigurationCache) resetConfiguration() {
+	c.currentConfiguration = c.localConfiguration
+	for _, rev := range c.orderedRevisions {
+		conf := c.enabledRevisions[rev]
+		c.updateConfiguration(*conf.enabled, conf.enabledNamespaces, conf.configID, conf.rcVersion)
+	}
+}
+
+func (c *instrumentationConfigurationCache) updateConfiguration(enabled bool, enabledNamespaces *[]string, rcID string, rcVersion int) Response {
+	log.Debugf("Updating current APM Instrumentation configuration")
```

Comment:
I think these 2 should be in the same line. Otherwise, others can get in between and it might be difficult to debug.